### DISCUSSION TITLE
Coloring by PLDDT

### DISCUSSION THREAD

**Discussion by mzhang5579** [molstar/molstar#848] [Q&A]
Hi all,

I was wondering if there's a way to color proteins by their plddt score per residue specifically for **local pdb files** instead of AlphaFold files.

I have my own pdb files with the plddt score in the beta column (same as AlphaFold pdb files). Uploading my pdb files works fine, but can't figure out how to color based on plddt score the way that AlphaFold does with their Model Confidence ranges (90-100 dark blue, 70-90 cyan, 50-70 yellow, 0-50 orange).

**Comment by Ieremie** [comment]
Plot by atom property-disorder

**Comment by 0gust1** [comment]
Hi, imho you should give some context from your usage, is it through the viewer app or through a lower-level of Molstar usage.

Disclaimer: I'll assume you use the viewer app, and I don't have the solution, but can give some leads/further info.

- The current Molstar viewer app already does that natively (and without any extra work), but for `cif` files only.  
- You're right, the pdb file from Alphafold won't show the plddt (and it's too bad, as Molstar, when loading a pdb file, internally convert it to CIF).
- When Molstar converts a PDB file to mmcif, the beta column/b-factor data ends up in a `B_iso_or_equiv` field, like in CIF file, 
 
So, as far as I understand, everything should be ok until now.

Now, let's see how the viewer loads and colors CIF files from alphafold DB:

it's done via: 
- [dedicated loadAlphaFoldDb function](https://github.com/molstar/molstar/blob/fb16cd0070846fa4e42121961f888d5e73d3876c/src/apps/viewer/app.ts#L303)
- the coloring is done by [QualityAssessmentPLDDTPreset](https://github.com/molstar/molstar/blob/fb16cd0070846fa4e42121961f888d5e73d3876c/src/apps/viewer/app.ts#L15C10-L15C38) that is registred in the viewer.

**But,** when you load an alphafold **pdb**, the PLDDT preset is not available in the Component > Preset toolbar of the Viewer.  
(see screenshots below)

**Temporary hypothesis:**
It seems that there is something that prevent the `QualityAssessmentPLDDTPreset` to be applicable to the loaded PDB file (even if the preset is registred by Molstar). The final answer may lie beneath the `loadAlphaFoldDb` function call and its dependencies: something which is done in this case and not the others **and/or** the behavior of the `isApplicable` function of the Preset. 

**AlphaFold PDB file:**
![AlphaFold PDB file](https://github.com/molstar/molstar/assets/578154/4baa2324-37dd-4c1f-907f-8633b79b650f)

**AlphaFold CIF file:**
![AlphaFold CIF file](https://github.com/molstar/molstar/assets/578154/6fb9eab9-e06f-4c60-8261-569ce1c2043b)
